crates/comments/src/directive.rs

```diff
@@ -30,15 +30,21 @@ pub enum FormatDirective {
 ///
 /// Note that directives are applied to the node they are attached to.
 pub fn parse_comment_directive(text: &str) -> Option<Directive> {
-    let text = text.strip_prefix('#')?;
+    let text = text.trim_start_matches('#');
     let text = text.trim_start();
-    let (category, text) = text.split_once(':')?;
-    let text = text.trim();
 
-    match category {
-        "fmt" => parse_format_directive(text),
-        _ => None,
+    if let Some((category, text)) = text.split_once(':') {
+        let text = text.trim();
+
+        #[allow(clippy::single_match)]
+        match category {
+            "fmt" => return parse_format_directive(text),
+            _ => {} // fallthrough
+        }
     }
+
+    let text = text.trim();
+    parse_special_directive(text)
```

Comment:
This feels a bit too awkward, I'm not really sure I understand the purpose of the fallthrough

I think this allows weirdness like `# weird: Generated by`

I feel like we can probably simplify this somehow?

Maybe it's

```
let text = text.strip_prefix('#')?;
let text = text.trim_start_matches('#');
let text = text.trim_start();

if let Some(directive) = parse_special_directive(text) {
  return Some(directive);
}

let (category, text) = text.split_once(':')?;
let text = text.trim();

match category {
        "fmt" => parse_format_directive(text),
        _ => None,
}
```

Reply:
We've cleared this up one-on-one